### PR TITLE
Ability to specify which TimePickerEntryModes to use and hide switch button

### PR DESCRIPTION
**Pull request by viralkachhadiya**
Related to #85013

added new two enum `dialOnly` and `inputOnly`. when App writer choose this type of `TimePickerEnteryMode`, showing only that type of dialog and hide switch button. when App writer choose other mode, showing switch button .

### CODE REVIEW

**Comment by viralkachhadiya** [comment]
@darrenaustin
@HansMuller 

 Can you please check my latest PR?

**Review by 3zrv — APPROVED** [review]

**Review by darrenaustin — CHANGES_REQUESTED** [review]
@viralkachhadiya thanks for the contribution. Looks good, but I have a few suggestions below.

In addition, we need a test to verify that this new feature works (and to prevent someone coming along later and breaking it 😉). You can add it to the rest of the time picker tests in: `packages/flutter/test/material/time_picker_test.dart`. If you need some help with this, you can look at the date picker PR that was mentioned in the bug report to see how that was tested.

**Comment by viralkachhadiya** [comment]
@darrenaustin  I resolved all issue and fixed with new structure. please checkout it.

**Comment by darrenaustin** [comment]
Hi @viralkachhadiya, looks like there is an issue when you merged master. It picked up a bunch of changes that weren't from you, which throws off the CLA bot. The easiest way to fix this is to rebase your changes on top of master and then force push to your branch on github:

```
% git pull --rebase upstream master
% git push --force origin specifying_timepickermode_switchbutton
```

File: packages/flutter/lib/src/material/time_picker.dart [hunk 1]
```diff
@@ -2015,6 +2021,10 @@ class _TimePickerDialogState extends State<TimePickerDialog> with RestorationMix
           _autofocusMinute.value = false;
           _entryMode.value = TimePickerEntryMode.dial;
           break;
+        case TimePickerEntryMode.dialOnly:
+        case TimePickerEntryMode.inputOnly:
+          assert(false, 'Can not change entry mode from _entryMode');
```
Comment (darrenaustin) [inline packages/flutter/lib/src/material/time_picker.dart:2026]: Instead of an assert it would be better to use a `FlutterError` here. Take a look for other examples that use this. Also, `_entryMode` in the string needs a `$` in front of it, or the error message will literarily be: `Can not change entry mode from _entryMode'.

**Review by darrenaustin — CHANGES_REQUESTED** [review]
Thanks for the updates, they look good (with a suggestion below). 

However we still need an added test for this feature as I mentioned in my last review.

**Comment by viralkachhadiya** [comment]
> Hi @viralkachhadiya, looks like there is an issue when you merged master. It picked up a bunch of changes that weren't from you, which throws off the CLA bot. The easiest way to fix this is to rebase your changes on top of master and then force push to your branch on github:
> 
> ```
> % git pull --rebase upstream master
> % git push --force origin specifying_timepickermode_switchbutton
> ```

i try it but still it has same merge issue.

**Comment by viralkachhadiya** [comment]
@darrenaustin I did not know. Why is it happing?
Can you help me please?

**Comment by viralkachhadiya** [comment]
@darrenaustin  can you check it?

**Comment by viralkachhadiya** [comment]
@PixelToast @Piinks can you please review this pr?

File: bin/internal/flutter_plugins.version [hunk 2]
```diff
@@ -1 +1 @@
-9bd9ca1d7423f1e1a1f063ae87deff41c2b191b1
+24febdf3aefdf469f7e30168a57f227682373146
```
Comment (darrenaustin) [inline bin/internal/flutter_plugins.version:1]: What does this change have to do with the TimePicker? It looks like something that came in with a merge that shouldn't have. Please revert this change.

**Review by darrenaustin — CHANGES_REQUESTED** [review]
@viralkachhadiya, sorry this took me a while to take another look at this. The code looks fine as I mentioned in my last review (although it looks like you have a new change that came in from a merge that will need to be reverted).

 However, you keep requesting reviews for this change when you have not yet provided a test for that ensures this change works. I have asked now twice before, once with pointers on how to go about writing such a test. Without a test we will not be able to accept this PR as per https://github.com/flutter/flutter/wiki/Tree-hygiene#tests.

File: bin/internal/flutter_plugins.version (hunk 2 above)
Comment (viralkachhadiya) [reply bin/internal/flutter_plugins.version]: i can't do it bcz whenever I merge my local branch with master then it is generate automatically

**Comment by viralkachhadiya** [comment]
@darrenaustin  I added test in test_picker.dart. please review my PR.

**Comment by Piinks** [comment]
Hi @viralkachhadiya it looks like many tests are failing currently for this PR, can you take a look?

**Comment by Piinks** [comment]
This PR has not been updated in a while, so I am going to close it. If you would like to continue working on this change, please re-open it. Thank you!
